src/abstractmcmc.jl

```diff
@@ -6,7 +6,7 @@ function AbstractMCMC.step(
     kwargs...,
 )
     logdensity = model.logdensity
-    logdensity = LogDensityProblemsAD.ADgradient(model.logdensity)
+    logdensity = LogDensityProblemsAD.ADgradient(:ForwardDiff, logdensity)
```

Comment:
You should be able to just remove this line if working with Turing.jl.

Now `model.logdensity` should already be wrapped in a `ADgradient`, and you can specify which one using the `adtype` kwarg:
```julia
externalsampler(sampler; adtype)
```

More generally, you should probably check if the model already has AD capabilties before wrapping it in `ADgradient`. Specifically, check `LogDensityProblems.capabilities(logdensity)` and make sure it's higher than 0-th order.